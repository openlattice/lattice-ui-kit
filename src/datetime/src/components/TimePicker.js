import React from 'react';
import { mergeDeep } from 'immutable';
import { TimePicker } from '@atlaskit/datetime-picker';
import selectStyles from '../../../style/selectStyles';

export const defaultProps = {
  selectProps: { styles: selectStyles },
  timeIsEditable: true
};

const LatticeTimePicker = (props) => {
  const mergedProps = mergeDeep(defaultProps, props);
  /* eslint-disable react/jsx-props-no-spreading */
  return <TimePicker {...mergedProps} />;
  /* eslint-enable */
};

export default LatticeTimePicker;
