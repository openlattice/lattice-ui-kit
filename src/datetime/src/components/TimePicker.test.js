import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TimePicker from './TimePicker';

describe('TimePicker', () => {

  it('render matches snapshot', () => {
    const tree = shallow(<TimePicker />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('clicking should toggle menu', () => {
    const wrapper = mount(<TimePicker />);
    expect(wrapper.find('Menu').exists()).toBeFalsy();
    wrapper.find('DropdownIndicator').simulate('mouseDown', { button: 0 });
    expect(wrapper.find('Menu').exists()).toBeTruthy();
  });

});
