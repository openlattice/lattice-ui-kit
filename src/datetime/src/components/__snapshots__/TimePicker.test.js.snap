// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePicker render matches snapshot 1`] = `
<WithAnalyticsContext(WithAnalyticsEvents(TimePicker))
  selectProps={
    Object {
      "styles": Object {
        "clearIndicator": [Function],
        "container": [Function],
        "control": [Function],
        "dropdownIndicator": [Function],
        "indicatorSeparator": [Function],
        "indicatorsContainer": [Function],
        "menu": [Function],
        "menuPortal": [Function],
        "option": [Function],
        "singleValue": [Function],
      },
    }
  }
  timeIsEditable={true}
/>
`;
